pyigm/guis/igmguesses.py

```diff
@@ -1634,31 +1659,62 @@ def remove_item(self,comp_name):
 
 def create_component(z, wrest, linelist, vlim=[-300.,300]*u.km/u.s,
                      spec=None):
+    """
+    Parameters
+    ----------
+    z
+    wrest
+    linelist
+    vlim
+    spec
+
+    Returns
+    -------
+
+    """
     # Transitions
     all_trans = linelist.all_transitions(wrest)
     if isinstance(all_trans, dict):
-        all_trans = [all_trans]
+        all_wrest = [all_trans['wrest']]
+    else:
+        if len(all_trans) == 1:
+            all_wrest = [all_trans['wrest']]
+        else:
+            all_wrest = Quantity(all_trans['wrest'])
     abslines = []
     if spec is not None:
         wvmin, wvmax = spec.wvmin, spec.wvmax
     else:
         wvmin, wvmax = 0.*u.AA, 1e9*u.AA
-    for trans in all_trans:
+    for iwrest in all_wrest:
+        #QtCore.pyqtRemoveInputHook()
+        #pdb.set_trace()
+        #QtCore.pyqtRestoreInputHook()
         # Restrict to those with spectral coverage!
-        if (trans['wrest'] * (1 + z) < wvmin) or (trans['wrest'] * (1 + z) > wvmax):
+        if (iwrest * (1 + z) < wvmin) or (iwrest * (1 + z) > wvmax):
             continue
-        aline = AbsLine(trans['wrest'],  linelist=linelist, z=z)
+        aline = AbsLine(iwrest, linelist=linelist, z=z)
         aline.limits.set(vlim)
         abslines.append(aline)
     if linelist.list != 'H2':
         if (abslines[0].data['Ej'].value > 0.):
             stars = '*'*(len(abslines[0].name.split('*'))-1)
         else:
             stars = None
-    else:
-        stars = None
-    # AbsComponent
-    comp = AbsComponent.from_abslines(abslines, stars=stars)
+        # AbsComponent
+        comp = AbsComponent.from_abslines(abslines, stars=stars)
+    else:  # H2
+        Zion = (-1, -1)  # code for molecules
+        Ntuple = (17, -1, 1)  # initial guess for Ntuple (needs to be given for adding lines from linelist; see below)
+        comp = AbsComponent(zero_coord, Zion, z, vlim, Ntup=Ntuple)
+        # add handy attrib
+        comp.attrib['init_wrest'] = wrest
+        # add abslines
+        comp.add_abslines_from_linelist(llist='H2', wvlim=(wvmin, wvmax), min_Wr=None)
+        # QtCore.pyqtRemoveInputHook()
```

Comment:
I also added this bit from a while ago (I had forgotten about this commit), but seems harmless here (only applies for LineList H2).